lib/CL/pocl_build.c

```diff
@@ -912,8 +912,7 @@ compile_and_link_program(int compile_program,
           goto ERROR;
         }
 
-      if (program->builtin_kernel_names == NULL)
-        setup_device_kernel_hashes (program);
+      setup_device_kernel_hashes (program);
```

Comment:
Does this also work with programs that don't actually compile their BI kernels at all? E.g. AlmaIF emulation device BI kernels? Is e.g. the build_hash[] still initialized to something reasonable for those?